src/test/scala/firrtlTests/transforms/GroupComponentsSpec.scala

```diff
@@ -0,0 +1,290 @@
+package firrtlTests
+package transforms
+
+import firrtl.annotations.{CircuitName, ComponentName, ModuleName}
+import firrtl.transforms.{GroupAnnotation, GroupComponents}
+
+class GroupComponentsSpec extends LowTransformSpec {
+  def transform = new GroupComponents()
+  val top = "Top"
+  def topComp(name: String): ComponentName = ComponentName(name, ModuleName(top, CircuitName(top)))
+  "The register r" should "be grouped" in {
+    val input =
+    s"""circuit $top :
+        |  module $top :
+        |    input clk: Clock
+        |    input data: UInt<16>
+        |    output out: UInt<16>
+        |    reg r: UInt<16>, clk
+        |    r <= data
+        |    out <= r
+      """.stripMargin
+    val groups = Seq(
+      GroupAnnotation(Seq(topComp("r")), "MyReg", "rInst", Some("_OUT"), Some("_IN"))
```

Comment:
Say if I have module A which instantiates modules C1 and C2 and also (unrelated) module B which instantiates modules C1 and C2 as well - does this pass handle that in one pass, or would I just use 2 separate GroupAnnotations and try to de-dup later?

Reply:
Sorry which module has the components to group, in this example?

Reply:
Sorry - my comment was more general and not tied to this specific example. I guess I'm just asking in the case of grouping the same thing across distinct modules.

Reply:
Grouping can only occur within a module, and the name of that module will not change. So, any all instances of the old module become instances of the now-split module.